### PR TITLE
Vel shift spectra marco

### PR DESCRIPTION
**Pull request by marcoleonardi97**
## Issue Type

- Bug
- Document
- Enhancement

## Checklist
- [x] I have read the [CONTRIBUTING.md]() -->
- [x] I have added docstrings to all methods
- [x] I have added sufficient comments to all lines
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no pep8 errors
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by marcoleonardi97** [comment]
renamed the flag to vel_shift and added it in a few more places, i'm sure it's still missing in some function entry but it's mainly done. also redirected to compute_particle_seds if vel_shift in stars and black holes, will it work like that?.

File: src/synthesizer/particle/stars.py [hunk 1]
```diff
@@ -566,8 +567,11 @@ def generate_lnu(
         )
 
         # Get the integrated spectra in grid units (erg / s / Hz)
-
-        lnu_particle = compute_integrated_sed(*args. shift)  # this hass not been modified yet, wont work as is, but shouldnt be hard to make it as compute_particle_sed
+        
+        if vel_shift:
+            lnu_particle = compute_particle_seds(*args, vel_shift) # ?
```
Comment (WillJRoper) [inline src/synthesizer/particle/stars.py:572]:
```
lnu_particle = compute_particle_seds(*args, vel_shift)
lnu_particle = np.sum(lnu_particle, axis=0)
```

shapes: (Npart, Nlam) -> (Nlam)

**Comment by marcoleonardi97** [comment]
I tried compiling and saw a bunch of errors that i'll go over this week. Most notably I made c a const double argument in the backend function but i forgot to pass it as a parameter in the front end, some funky function declaration in particle_spectra.c and possibly a missing } or some ; in some lines.

**Comment by marcoleonardi97** [comment]
passed c as an argument wherever it was needed (hopefully) and defaulted it to 2.998e8 m/s in the front end (just so existing code using generate_lnu without vel_shift doesn't stop working)

**Pull request merged**